### ISSUE TITLE
.

### ISSUE THREAD

**Issue by Lhy142857** [HZ-LL/MA-UNet#2]
For the third dataset, how should the images and masks in the training and testing sets be placed? If all the images are placed in the "images" folder and all the masks are placed in the "masks" folder, when running Ktrain.py, an error will occur stating that the test.tif image file cannot be found in the "masks" folder. Moreover, the format of the test.tif image file is that of an image in the "images" folder, not the format of the images in the "masks" folder. Please provide an explanation. Thank you.

**Comment by HZ-LL** [comment]
Perhaps due to the inconsistency of image names in "images" and "masks", the training is indexed by the image name. So you have to make sure that the image is named the same in order to index the image and its label correctly.

**Comment by Lhy142857** [comment]
If the image names in both the "images" and "masks" are exactly the same, an error will occur when attempting to load the images, preventing the images from being loaded.

**Comment by HZ-LL** [comment]
My data is stored in a format like "01.tif", about 40 images. I just tested locally and didn't have any issues. Maybe you can leave the error to ChatGPT to deal with. If there are no problems with the other two datasets, the third dataset should not be a problem either. Double-check issues such as dataset formatting.

**Comment by Lhy142857** [comment]
Are the images stored in the ".tif" format in the "images" folder and in the ".gif" format in the "masks" folder, or are they both stored in the ".tif" format?

**Comment by HZ-LL** [comment]
Both folders have the same data name and format












李洪智
***@***.***






---Original---
From: ***@***.***&gt;
Date: Fri, Mar 28, 2025 10:29 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [HZ-LL/MA-UNet] . (Issue #2)


   
Are the images stored in the ".tif" format in the "images" folder and in the ".gif" format in the "masks" folder, or are they both stored in the ".tif" format?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID: ***@***.***&gt;
  Lhy142857 left a comment (HZ-LL/MA-UNet#2)
 
Are the images stored in the ".tif" format in the "images" folder and in the ".gif" format in the "masks" folder, or are they both stored in the ".tif" format?

**Comment by HZ-LL** [comment]
Both folders have the same data name and format.If the image format is different, you need to convert it first.

**Comment by Lhy142857** [comment]
Hello, I'd like to ask that when I run the DDTL dataset alone, I set the learning rate to 0.00001, and as a result, the AUC indicator is approximately 10% higher than that presented in the article. Regarding this phenomenon, could it be caused by the reduction of the learning rate? Thank you.

**Comment by HZ-LL** [comment]
This is possible because the IoU trend chart of the article shows that the MA-UNet is not converging on the DDTI. Also inspired by you, I found an issue with the cosine annealing scheduler parameters while sorting out the code, I lowered the minimum loss and found it worked really well (see result/IoU_Index_DDTI.png), after which I will update the code version and the new pre-training weights. Thank you so much for inspiring me.

**Comment by Lhy142857** [comment]
When running the third dataset alone, my results exceed the original paper's by: Dice +20%, IoU +30%, Precision +21%, Recall +10%, AUC +10%. Could this be the same issue as before? If so, would you mind adjusting the code? Thanks a lot for your support!
